### PR TITLE
Added Login back

### PR DESCRIPTION
**Pull request by bjrne**
after it disappeared with the new navigationControllers. The login Button is now visible in almost every Tab. It's not in the course list tab, because it looks really bad with the segmented control und if you're not logged in and try to click on a course, you're asked to login anyway.

### CODE REVIEW

File: xikolo-ios/AbstractTabContentViewController.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    var isLoggedIn = UserProfileHelper.isLoggedIn()
+
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self)
+    }
+
+    private struct Constants{
+        static let ShowLoginSegue = "ShowLoginFromBarButton"
+    }
```
Comment (invliD) [inline xikolo-ios/AbstractTabContentViewController.swift]: Why put this in a private struct?

File: xikolo-ios.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -47,6 +47,8 @@
 		24895DC31CCC09C700D6DFB8 /* ImageHelper.swift in Sources */ = {isa = PBXBuildFile; fileRef = 24895DC11CCC09C300D6DFB8 /* ImageHelper.swift */; };
 		2490C4C71D1C158B00F42949 /* MainViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2490C4C61D1C158B00F42949 /* MainViewController.swift */; };
 		2490C4C91D1C486100F42949 /* CourseHeaderView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2490C4C81D1C486100F42949 /* CourseHeaderView.swift */; };
+		2490C4CB1D1C5A9900F42949 /* Settings.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2490C4CA1D1C5A9900F42949 /* Settings.swift */; };
+		2490C4CF1D1C5F3600F42949 /* Settings.swift in Sources */ = {isa = PBXBuildFile; fileRef = 2490C4CA1D1C5A9900F42949 /* Settings.swift */; };
```
Comment (invliD) [inline xikolo-ios.xcodeproj/project.pbxproj]: Please remove these 4 useless changes that move Settings.swift around.

File: xikolo-ios/AbstractTabContentViewController.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+    func updateLogoutState() {
+        isLoggedIn = false
+        updateUIAfterLoginLogoutAction()
+    }
+
+    func updateUIAfterLoginLogoutAction() {
+        if isLoggedIn {
+            dispatch_async(dispatch_get_main_queue(), {
+                self.navigationItem.rightBarButtonItem?.title = ""
```
Comment (invliD) [inline xikolo-ios/AbstractTabContentViewController.swift]: wtf?

Comment (bjrne) [inline xikolo-ios/AbstractTabContentViewController.swift]: Could you please be a bit more specific with your thoughts on this?

Comment (bjrne) [inline xikolo-ios/AbstractTabContentViewController.swift]: Because i couldn't find the "hidden" attribute with a BarButton. I'll change it to nil then?

Comment (invliD) [inline xikolo-ios/AbstractTabContentViewController.swift]: This is a lot cleaner: http://stackoverflow.com/questions/10021748/how-do-i-show-hide-a-uibarbuttonitem

File: xikolo-ios/AbstractTabContentViewController.swift [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        isLoggedIn = false
+        updateUIAfterLoginLogoutAction()
+    }
+
+    func updateUIAfterLoginLogoutAction() {
+        if isLoggedIn {
+            dispatch_async(dispatch_get_main_queue(), {
+                self.navigationItem.rightBarButtonItem = nil
+
+            })
```
Comment (invliD) [inline xikolo-ios/AbstractTabContentViewController.swift]: Isn't this function called on the main thread anyway?

Comment (bjrne) [inline xikolo-ios/AbstractTabContentViewController.swift]: Most of the time it's triggered by a notification and I had problems with it once, so I added the line. But i tested it now without it and it seems fine. If you're sure that it's not a problem, I will remove it.

**Pull request merged**